frontend/src/scenes/error-tracking/utils.ts

```diff
@@ -128,11 +128,89 @@ export function hasAnyInAppFrames(exceptionList: ErrorTrackingException[]): bool
     return exceptionList.some(({ stacktrace }) => stacktrace?.frames?.some(({ in_app }) => in_app))
 }
 
-export const sparklineLabelsDay = sparklineLabels({ value: 24, interval: 'hour' })
-export const sparklineLabelsMonth = sparklineLabels({ value: 31, interval: 'day' })
+export function generateSparklineLabels(range: DateRange, resolution: number): string[] {
+    const resolvedDateRange = resolveDateRange(range)
+    const from = dayjs(resolvedDateRange.date_from)
+    const to = dayjs(resolvedDateRange.date_to)
+    const labels = Array.from({ length: resolution }, (_, i) => {
+        const bin_size = Math.floor(to.diff(from, 'seconds') / resolution)
+        return from.add(i * bin_size, 'seconds').toISOString()
+    })
+    return labels
+}
+
+export class ResolvedDateRange {
+    date_from: Dayjs
+    date_to: Dayjs
+
+    constructor(date_from: Dayjs, date_to: Dayjs) {
+        this.date_from = date_from
+        this.date_to = date_to
+    }
+
+    toDateRange(): DateRange {
+        return {
+            date_from: this.date_from.toISOString(),
+            date_to: this.date_to.toISOString(),
+        }
+    }
+
+    static fromDateRange(dateRange: DateRange): ResolvedDateRange {
+        return new ResolvedDateRange(resolveDate(dateRange.date_from), resolveDate(dateRange.date_to))
+    }
+}
+
+// Converts relative date range to absolute date range
+export function resolveDateRange(dateRange: DateRange): ResolvedDateRange {
+    return ResolvedDateRange.fromDateRange(dateRange)
+}
+
+// Converts relative date to absolute date.
+export function resolveDate(date?: string | null): Dayjs {
+    if (!date) {
+        return dayjs()
+    }
+    if (date == 'all') {
+        return dayjs().subtract(1, 'year')
+    }
+    const parsedDate = datetimeStringToDayJs(date)
+    if (parsedDate) {
+        return parsedDate
+    }
+    throw new Error(`Invalid date: ${date}`)
+}
```

Comment:
It feels a little complicated to me adding the class given we always call `toDateRange`. Why not just make all this 2 simple methods?
```suggestion
export function resolveRelativeDateRange(dateRange: DateRange): DateRange {
    return {
        date_from: resolveRelativeDate(dateRange.date_from).toISOString(),
        date_to: resolveRelativeDate(dateRange.date_to).toISOString(),
    }
}

export function resolveRelativeDate(date?: string | null): Dayjs {
    if (!date) {
        return dayjs()
    }
    if (date == 'all') {
        return dayjs().subtract(1, 'year')
    }
    const parsedDate = datetimeStringToDayJs(date)
    if (parsedDate) {
        return parsedDate
    }
    throw new Error(`Invalid date: ${date}`)
}
```

Reply:
It's easier to keep the date range as dayjs object as long as possible. This I can construct labels, format dates on the UI without having the parse the date all the time.